File: src/main/java/com/itsci/mju/maebanjumpen/transaction/service/TransactionService.java
package com.itsci.mju.maebanjumpen.transaction.service;

import com.fasterxml.jackson.databind.JsonNode;
import com.itsci.mju.maebanjumpen.transaction.dto.TransactionDTO;
import java.util.List;
import java.util.Optional;

public interface TransactionService {

    // Basic CRUD operations: ใช้ DTO
    List<TransactionDTO> getAllTransactions();
    Optional<TransactionDTO> getTransactionById(int id);
    TransactionDTO saveTransaction(TransactionDTO transactionDto);
    void deleteTransaction(int id);

    // Member-specific operations: ใช้ DTO
    List<TransactionDTO> getTransactionsByMemberId(int memberId);

    // Withdrawal-related operations: ใช้ DTO
    List<TransactionDTO> getWithdrawalRequests();
    Optional<TransactionDTO> updateWithdrawalRequestStatus(Integer transactionId, String newStatus);

    // Omise Webhook Processing: ⬅️ เมธอดสำหรับการประมวลผล Webhook
    void processOmiseChargeComplete(JsonNode omisePayload) throws Exception;


    // Status-related utility methods
    String getLocalizedStatus(String status, boolean isEnglish);
    String getStatusColorHex(String status);
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/transaction/service/TransactionService.kt
package com.itsci.mju.maebanjumpen.transaction.service

import com.fasterxml.jackson.databind.JsonNode
import com.itsci.mju.maebanjumpen.transaction.dto.TransactionDTO
import java.util.Optional

interface TransactionService {
    fun getAllTransactions(): List<TransactionDTO>
    fun getTransactionById(id: Int): Optional<TransactionDTO>
    fun saveTransaction(transactionDto: TransactionDTO): TransactionDTO
    fun deleteTransaction(id: Int)
    fun getTransactionsByMemberId(memberId: Int): List<TransactionDTO>
    fun getWithdrawalRequests(): List<TransactionDTO>
    fun updateWithdrawalRequestStatus(transactionId: Int, newStatus: String): Optional<TransactionDTO>
    fun processOmiseChargeComplete(omisePayload: JsonNode)
    fun getLocalizedStatus(status: String?, isEnglish: Boolean): String
    fun getStatusColorHex(status: String?): String
}

